python/ray/serve/tests/test_api.py

```diff
@@ -774,6 +774,49 @@ def check_missing_required(args: self.TypedArgs):
                 check_missing_required, {"num_replicas": "10"}
             )
 
+    def test_pydantic_version_compatibility(self):
+        """Check compatibility with different pydantic versions."""
+
+        cat_dict = {"color": "orange", "age": 10}
+
+        from pydantic import BaseModel
+
+        class Cat(BaseModel):
+            color: str
+            age: int
+
+        def build(args: Cat):
+            """Builder with Pydantic model type hint."""
+
+            assert isinstance(args, Cat), f"args type: {type(args)}"
+            assert args.color == cat_dict["color"]
+            assert args.age == cat_dict["age"]
+            return self.A.bind(f"My {args.color} cat is {args.age} years old.")
+
+        app = call_app_builder_with_args_if_necessary(build, cat_dict)
+        assert isinstance(app, Application)
+
+        try:
```

Comment:
Non-blocker: This can probably DRY up with pytest parametrize 🙃

Reply:
Good call, I parametrized it.